components/instructions/ExecuteAllInstructionButton.tsx

```diff
@@ -0,0 +1,147 @@
+import { useEffect, useState } from 'react'
+import {
+  InstructionExecutionStatus,
+  Proposal,
+  ProposalTransaction,
+  ProposalState,
+} from '@solana/spl-governance'
+import React from 'react'
+import { CheckCircleIcon, PlayIcon, RefreshIcon } from '@heroicons/react/solid'
+import Button from '@components/Button'
+import { RpcContext } from '@solana/spl-governance'
+import useRealm from '@hooks/useRealm'
+import useWalletStore from 'stores/useWalletStore'
+import { ProgramAccount } from '@solana/spl-governance'
+import { PublicKey } from '@solana/web3.js'
+import Tooltip from '@components/Tooltip'
+import { getProgramVersionForRealm } from '@models/registry/api'
+import { executeInstructions } from 'actions/executeInstructions'
+
+export enum PlayState {
+  Played,
+  Unplayed,
+  Playing,
+  Error,
+}
+
+export function ExecuteAllInstructionButton({
+  proposal,
+  playing,
+  setPlaying,
+  proposalInstructions,
+}: {
+  proposal: ProgramAccount<Proposal>
+  proposalInstructions: ProgramAccount<ProposalTransaction>[]
+  playing: PlayState
+  setPlaying: React.Dispatch<React.SetStateAction<PlayState>>
+}) {
+  const { realmInfo } = useRealm()
+  const wallet = useWalletStore((s) => s.current)
+  const connection = useWalletStore((s) => s.connection)
+  const fetchRealm = useWalletStore((s) => s.actions.fetchRealm)
+  const connected = useWalletStore((s) => s.connected)
+
+  const [currentSlot, setCurrentSlot] = useState(0)
+
+  const canExecuteAt = proposal?.account.votingCompletedAt
+    ? proposal.account.votingCompletedAt.toNumber() + 1
+    : 0
+
+  const ineligibleToSee = currentSlot - canExecuteAt >= 0
```

Comment:
I don't find `ineligibleToSee` to be a very descriptive name, can we find a better one?

Reply:
`isPassedExecutionSlot` ?